backend/src/main/java/atlantbh/restaurants/repositories/BaseRepository.java

```diff
@@ -0,0 +1,20 @@
+package atlantbh.restaurants.repositories;
+
+import atlantbh.restaurants.exceptions.RepositoryException;
+import atlantbh.restaurants.models.filters.BaseFilterBuilder;
+
+import java.util.List;
+
+public interface BaseRepository<T, S extends Enum<S>, F extends BaseFilterBuilder<S, F>> {
+    void delete(Long id) throws RepositoryException;
+
+    T findById(Long id) throws RepositoryException;
+
+    T save(T entity) throws RepositoryException;
+
+    T update(T entity) throws RepositoryException;
+
+    Long count(F filterBuilder);
+
+    List<T> findAll();
```

Comment:
this should also contain 
```
PaginatedResult<T> find(F filterBuilder);
```

Reply:
Fixed 8c841a20e489a31c66d177c8682339f9d4f19de0